allegro.h

```diff
@@ -580,11 +580,15 @@ typedef class Serial_read_buffer : public Serial_buffer {
     double get_double() { assert(!(((intptr_t) ptr) & 7));
                           double d = *((double *) ptr); ptr += sizeof(double); 
                           return d; }
-    const char *get_string() { char *s = ptr; char *fence = buffer + len;
-                         assert(ptr < fence);
-                         while (*ptr++) assert(ptr < fence);
-                         get_pad();
-                         return s; }
+    const char *get_string() {
+        char *s = ptr;
+        char *fence = buffer + len;
+        assert(ptr < fence);
```

Comment:
I think the right thing here is, since fence is only used for parameter validation and assert only works when NDEBUG is undefined, lines 585 thru 588 should be inside #ifndef NDEBUG ... #endif (and then delete line 588)

Reply:
Maybe I'm confused by github. Why is this marked resolved?

Reply:
Yes, I'm really confused by Github, which is showing the same conversation in 2 different states in 2 different tabs right now.  I'll switch to what seems to be newer.

Reply:
Github is acting weird, this is not showing up-to-date code. If you look at the diff in the 'Files Changed' tab it shows the current state of the PR.